terraform_token.go

```diff
@@ -96,23 +94,34 @@ func (b *tfBackend) terraformTokenRevoke(ctx context.Context, req *logical.Reque
 		}
 	}
 
-	teamID := ""
-	teamIDRaw, ok := req.Secret.InternalData["team_id"]
-	if ok {
-		teamID, ok = teamIDRaw.(string)
-		if !ok {
-			return nil, fmt.Errorf("secret is missing team_id internal data")
+	if isOrgToken(organization, teamID) {
+		// revoke org API token
+		if err := client.OrganizationTokens.Delete(ctx, organization); err != nil {
+			return nil, fmt.Errorf("error revoking organization token: %w", err)
 		}
+		return nil, nil
 	}
 
-	var data = map[string]interface{}{
-		"organization": organization,
-		"token_id":     tokenID,
-		"team_id":      teamID,
+	if isTeamToken(teamID) {
+		// revoke team API token
+		if err := client.TeamTokens.Delete(ctx, teamID); err != nil {
+			return nil, fmt.Errorf("error revoking team token: %w", err)
+		}
+		return nil, nil
 	}
 
-	if err := b.walRollback(ctx, req, terraformTokenType, data); err != nil {
-		return nil, err
+	// if we haven't returned yet, then the token is a user API token
+	tokenID := ""
+	tokenIDRaw, ok := req.Secret.InternalData["token_id"]
+	if ok {
+		tokenID, ok = tokenIDRaw.(string)
+		if !ok {
+			return nil, fmt.Errorf("secret is missing tokenID internal data")
+		}
+	}
+
```

Comment:
You can reach to this point and have `team_id`, `organization`, `token_id` all be present but empty string values (the `ok` checks only validates presence and not whether it's the zero-value). We should also make sure that `tokenID` is not empty before proceeding.

Reply:
Done in 1d8037d